extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/user/UserService.java

```diff
@@ -88,6 +88,7 @@ private void putAllUsers(Map<String, User> users, LDAPConnection ldapConnection,
             // Set search limits
             LDAPSearchConstraints constraints = new LDAPSearchConstraints();
             constraints.setMaxResults(confService.getMaxResults());
+            constraints.setDereference(confService.getDereferenceAliases());
```

Comment:
Create getLDAPSearchConstraints in the ConfigurationService class which pulls in getMaxResults() and getDereferenceAliases() and sets up the constraints.  Changed code in UserService and ConnectionService to use this function from confService instead of setting up their own each time.